terraform/shared/opensearch/opensearch.tf

```diff
@@ -0,0 +1,111 @@
+resource "aws_ssm_parameter" "opensearch" {
+  name  = var.service_name
+  type  = "String"
+  value = local.connection_config
+
+  tags = merge(
+    var.tags,
+    {
+      Name = "parameter-storage"
+    }
+  )
+}
+
+resource "random_password" "es_password" {
+  length      = 16
+  special     = true
+  numeric     = true
+  upper       = true
+  lower       = true
+  min_lower   = 1
+  min_numeric = 1
+  min_special = 1
+  min_upper   = 1
+}
+
+resource "aws_security_group" "es" {
+  name        = "allow-es-${var.service_name}"
+  description = "Allow connection to Elasticsearch"
+  vpc_id      = var.rancher_vpc
+
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = merge(
+    var.tags,
+    {
+      Name = "allow-es"
+  })
+}
+
+module "aws_es" {
```

Comment:
As in `resource "aws_ssm_parameter" "opensearch"` let's use here also "opensearch" instead of es

Reply:
done